### PR TITLE
Support for pagespeed in vhosting configs

### PR DESCRIPTION
**Pull request by syphernl**
- **Depends on:** https://github.com/Enrise/nginx-formula/pull/31
- Make sure you re-run the tests after merging above

### CODE REVIEW

File: pillar.example [hunk 1]
```diff
@@ -85,6 +88,29 @@ vhosting:
...
+
+        # Pagespeed example with specific config
+        #pagespeed-example-config.com:
+        #  pagespeed:
+        #    enable: True
+        #    config:
+        #      DisableFilters: 'rewrite_images,combine_css'
+
+        # Add extra config to the location section (nginx only)
+        extra-location-config.com:
```
Comment (bobbybouwmann) [inline pillar.example:110]: Should this config thing also be in comments since it's an example?

File: vhosting/pagespeed.sls [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# Install pagespeed module
+{%- set webserver = salt['pillar.get']('vhosting:server:webserver', 'nginx') %}
+
+{%- if webserver == 'nginx' %}
+{%- set pkg_name = 'nginx-module-pagespeed' %}
```
Comment (bobbybouwmann) [inline vhosting/pagespeed.sls:5]: This might be irrelevant or not even possible, but what about indentation here?

**Review by bobbybouwmann — COMMENTED** [review]
Looks good to me, only two points ;)

File: vhosting/pagespeed.sls (hunk 2 above)
Comment (syphernl) [reply vhosting/pagespeed.sls:5]: Jinja can be rather picky if you start indenting, that's why we don't for blocks like these.

File: pillar.example (hunk 1 above)
Comment (syphernl) [reply pillar.example:110]: Any vhost configs that are testable are enabled in this file. Only the one's that aren't possible such as LetsEncrypt (due to domain ownership check) and Pagespeed (due to missing module not available in the public repositories) are commented out to serve as an example. But technically the whole file _is_ an example :-)

Comment (bobbybouwmann) [reply pillar.example:110]: 👍

File: pillar.example [hunk 3]
```diff
@@ -85,6 +88,29 @@ vhosting:
...
+        #  pagespeed:
+        #    enable: True
+        #    file_cache_path: '/tmp/pagespeed'
+
+        # Pagespeed example with specific config
+        #pagespeed-example-config.com:
+        #  pagespeed:
+        #    enable: True
+        #    config:
+        #      DisableFilters: 'rewrite_images,combine_css'
```
Comment (nstapelbroek) [inline pillar.example:107]: Would it be possible to declare this as an array? since it is one :see_no_evil:

Comment (syphernl) [reply pillar.example:107]: Yes, I did have this before but found out that there are about 50 config keys. Handling them all is horrible, so therefor its inserting this rawly into the pagespeed section.

**Review by nstapelbroek — APPROVED** [review]
Discussed in person. Everything in the config is evaluated into the pagespeed config block.

**Review by eXistenZNL — APPROVED** [review]
Fine by me!

**Pull request merged**